src/components/tiles/table/use-columns-from-data-set.ts

```diff
@@ -96,8 +96,12 @@ export const useColumnsFromDataSet = ({
     }
     columnChanges;  // eslint-disable-line no-unused-expressions
     return cols;
-  }, [attributes, rowHeight, RowLabelHeader, RowLabelFormatter, readOnly, columnChanges,
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [attributes, attributes.length, rowHeight, RowLabelHeader, RowLabelFormatter, readOnly, columnChanges,
       ColumnHeaderCell, controlsColumn, cellClasses, measureColumnWidth, metadata, lookupImage]);
+  // attributes.length has been included above so the columns are recreated when columns are added or removed
+  // from external means (such as undo/redo). It would be better to make this hook observe changes to the model,
+  // but I'm not sure how to do that.
```

Comment:
There is an approach where you basically create a temporary MobX class inside of the component. There is a hook to help you do that which I used once. However in that case it ended up making more sense to make an actual MobX class with an instance that was created by a parent component and passed to the child component as a property. So you won't find that hook in the current CLUE codebase. I'd guess making some kind of MobX class with a shared instance would be the right way to go here too. In some systems these are called "presentation models". They are models encapsulating logic used by views which shouldn't be in the real model. https://martinfowler.com/eaaDev/PresentationModel.html